lib/msal-react/src/hooks/useAccount.ts

```diff
@@ -4,42 +4,28 @@
  */
 
 import { useState, useEffect } from "react";
-import { AccountInfo, IPublicClientApplication, InteractionStatus, AccountEntity } from "@azure/msal-browser";
+import { AccountInfo, IPublicClientApplication, AccountEntity } from "@azure/msal-browser";
 import { useMsal } from "./useMsal";
 import { AccountIdentifiers } from "../types/AccountIdentifiers";
+import { getAccountByIdentifiers } from "../utils/utilities";
 
-function getAccount(instance: IPublicClientApplication, accountIdentifiers: AccountIdentifiers): AccountInfo | null {
-    const allAccounts = instance.getAllAccounts();
-    if (allAccounts.length > 0 && (accountIdentifiers.homeAccountId || accountIdentifiers.localAccountId || accountIdentifiers.username)) {
-        const matchedAccounts = allAccounts.filter(accountObj => {
-            if (accountIdentifiers.username && accountIdentifiers.username.toLowerCase() !== accountObj.username.toLowerCase()) {
-                return false;
-            }
-            if (accountIdentifiers.homeAccountId && accountIdentifiers.homeAccountId.toLowerCase() !== accountObj.homeAccountId.toLowerCase()) {
-                return false;
-            }
-            if (accountIdentifiers.localAccountId && accountIdentifiers.localAccountId.toLowerCase() !== accountObj.localAccountId.toLowerCase()) {
-                return false;
-            }
-
-            return true;
-        });
-
-        return matchedAccounts[0] || null;
-    } else {
-        return null;
+function getAccount(instance: IPublicClientApplication, accountIdentifiers?: AccountIdentifiers): AccountInfo | null {
+    if (!accountIdentifiers || (!accountIdentifiers.homeAccountId && !accountIdentifiers.localAccountId && !accountIdentifiers.username)) {
+        // If no account identifiers are provided, return active account
+        return instance.getActiveAccount();
     }
+
+    return getAccountByIdentifiers(instance.getAllAccounts(), accountIdentifiers);
 }
 
 /**
  * Given 1 or more accountIdentifiers, returns the Account object if the user is signed-in
  * @param accountIdentifiers 
  */
-export function useAccount(accountIdentifiers: AccountIdentifiers): AccountInfo | null {
+export function useAccount(accountIdentifiers?: AccountIdentifiers): AccountInfo | null {
     const { instance, inProgress } = useMsal();
 
-    const initialStateValue = inProgress === InteractionStatus.Startup ? null : getAccount(instance, accountIdentifiers);
-    const [account, setAccount] = useState<AccountInfo|null>(initialStateValue);
+    const [account, setAccount] = useState<AccountInfo|null>(() => getAccount(instance, accountIdentifiers));
 
     useEffect(() => {
```

Comment:
Be careful with having the value you're updating in a hook be a dependency for the hook as well (accounts). It's usually better to use [functional updates](https://reactjs.org/docs/hooks-reference.html#functional-updates) when you need to compare or mutate the current state value. 

I think the current approach will end up re-running the useEffect hook until the value is stable. Also you might encounter cases where the current value is outdated, as setState only queues an update. With functional updates the value will be updated.